File: 448-find-all-numbers-disappeared-in-an-array/main.go
func findDisappearedNumbers(nums []int) []int {
    
    for _, num := range nums {
        idx := abs(num) - 1
        if nums[idx] > 0 {
            nums[idx] = -nums[idx]
        }
    }

    var missing []int
    for idx := range len(nums) {
        if nums[idx] > 0 {
            missing = append(missing, idx + 1)
        }
    }

    return missing
}

func abs(x int) int {
    if x < 0 {
        return -x
    }

    return x
}

File: 448-find-all-numbers-disappeared-in-an-array/main.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        
        _counter = Counter(nums)
        _numbers = [i + 1 for i in range(len(nums))]
        
        _missing = []
        for _number in _numbers:
            if _number not in _counter.keys():
                _missing.append(_number)

        return _missing
